python/pyspark/sql/pandas/conversion.py

```diff
@@ -0,0 +1,431 @@
+        finally:
+            # Join serving thread and raise any exceptions from collectAsArrowToPython
+            jsocket_auth_server.getResult()
+
+        # Separate RecordBatches from batch order indices in results
+        batches = results[:-1]
+        batch_order = results[-1]
+
+        # Re-order the batch list using the correct order
+        return [batches[i] for i in batch_order]
+
+
+class SparkConversionMixin(object):
+    """
+    Min-in for the conversion from pandas to Spark. Currently, only :class:`SparkSession`
+    can use this class.
+    """
+    def createDataFrame(self, data, schema=None, samplingRatio=None, verifySchema=True):
+        from pyspark.sql import SparkSession
+
+        assert isinstance(self, SparkSession)
+
+        from pyspark.sql.pandas.utils import require_minimum_pandas_version
+        require_minimum_pandas_version()
+
+        if self._wrapped._conf.pandasRespectSessionTimeZone():
+            timezone = self._wrapped._conf.sessionLocalTimeZone()
+        else:
+            timezone = None
+
+        # If no schema supplied by user then get the names of columns only
+        if schema is None:
+            schema = [str(x) if not isinstance(x, basestring) else
+                      (x.encode('utf-8') if not isinstance(x, str) else x)
+                      for x in data.columns]
+
+        if self._wrapped._conf.arrowPySparkEnabled() and len(data) > 0:
+            try:
+                return self._create_from_pandas_with_arrow(data, schema, timezone)
+            except Exception as e:
+                from pyspark.util import _exception_message
+
+                if self._wrapped._conf.arrowPySparkFallbackEnabled():
+                    msg = (
+                        "createDataFrame attempted Arrow optimization because "
+                        "'spark.sql.execution.arrow.pyspark.enabled' is set to true; however, "
+                        "failed by the reason below:\n  %s\n"
+                        "Attempting non-optimization as "
+                        "'spark.sql.execution.arrow.pyspark.fallback.enabled' is set to "
+                        "true." % _exception_message(e))
+                    warnings.warn(msg)
+                else:
+                    msg = (
+                        "createDataFrame attempted Arrow optimization because "
+                        "'spark.sql.execution.arrow.pyspark.enabled' is set to true, but has "
+                        "reached the error below and will not continue because automatic "
+                        "fallback with 'spark.sql.execution.arrow.pyspark.fallback.enabled' "
+                        "has been set to false.\n  %s" % _exception_message(e))
+                    warnings.warn(msg)
+                    raise
+        data = self._convert_from_pandas(data, schema, timezone)
+        return self._create_dataframe(data, schema, samplingRatio, samplingRatio)
```

Comment:
Why `..., samplingRatio, samplingRatio)`? Should it be `..., verifySchema, samplingRatio)`?

Reply:
Thanks for pointing this out. I made a PR - https://github.com/apache/spark/pull/27225